### PR TITLE
Creating World Weather mobile application.

### PR DESCRIPTION
**Pull request by ahzamir**
### In this pull request I have added changes to my react app according to the requirements:

- I used two APIs with numeric data for this project.

- I have used the original given design to give design to my app components.

- I have added three pages on my weather app:

  - Home page

    - When the page loads, the web app shows the list of categories that are filtered out by country names.
    - Along with the category name, I displayed states per category that come from the API.
    - When the user clicks (or taps) on a countries item, the application navigates to the states page.
    
  - Details page

    - On the states page, the web app retrieves data from the API to show weather data of the country and the states of the country.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.
    - When the users click on the states it goes to the states page.
    
  - States page
    - On the states page it shows the weather of the specified clicked state.
 
- Testing requirements

  - I have created unit tests for pure functions in my Redux code.
  - I have created integration tests for my application using the React Testing Library.
  - I mocked the access to the API.
  - I mocked the connection to the Redux Store.

### CODE REVIEW

File: src/components/pages/HomePage/Countries.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      <div className="padding countries">
+        {Object.values(countries).map((country) => (
+          <Country
+            key={country.iso3}
+            name={country.name}
+            states={country.states}
+            iso3={country.iso3}
+          />
+        ))}
+      </div>
```
Comment (ShadyShawkat) [inline src/components/pages/HomePage/Countries.js:30]: - [ ] Kindly stick to the original design as much as possible. Please make sure that the list of countries is displayed with 2 countries in each row, to match the original design.

| Your design | Original design |
| --- | --- |
| image | image |

File: src/components/pages/StatesPage/States.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      <div className="states padding">
+        {countryState.states.map((state) => (
+          <State
+            name={state.name}
+            key={state.name}
+            code={state.state_code}
+          />
+        ))}
+      </div>
+    </div>
```
Comment (ShadyShawkat) [inline src/components/pages/StatesPage/States.js:39]: - [ ] Kindly stick to the original design as much as possible. Please make sure that the list of states in each country is displayed one in each row, to match the original design.

| Your design | Original design |
| --- | --- |
| image | image |

File: README.md [hunk 3]
```diff
@@ -1,26 +1,29 @@
...
+- Webpack
+- Jest
 
-## Live Demo (if available) 💻
+## Live Demo (Netlify) 💻
 
-[Live Demo Link](https://livedemo.com)
+- [World Weather](https://development--gregarious-pavlova-aaa751.netlify.app/)
 
+- [Live Demo](https://www.loom.com/share/3b83c1bafd504fa09bbfc763cc2f3afc)
```
Comment (ShadyShawkat) [inline README.md]: - [ ] Please make sure to talk about a feature or a piece of code you're proud of, as per [this](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=you%20should%20also%20highlight%20some%20interesting%20piece%20of%20code%20or%20something%20you%20built%20that%20you%20are%20very%20proud%20of.) requirement

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @ahzamir, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All tests are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 
- Good overall design ✔️ 
- Your code is clean and well documented ✔️

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by RolandM99 — CHANGES_REQUESTED** [review]
Hi @ahzamir, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

- [ ] As requested by the previous reviewer, make sure that in your loom video presentation you highlighted a piece of code that you find interesting and you're proud of making. record a new one with changes implemented from the previous review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/pages/HomePage/Countries.js (hunk 1 above)
Comment (Whoistolu) [reply src/components/pages/HomePage/Countries.js:30]: Well resolved 👍🏽

File: src/components/pages/StatesPage/States.js (hunk 2 above)
Comment (Whoistolu) [reply src/components/pages/StatesPage/States.js:39]: Well resolved 👍🏽

File: README.md (hunk 3 above)
Comment (Whoistolu) [reply README.md]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @ahzamir 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahzamir** [comment]
@Whoistolu, @ShadyShawkat, @RolandM99  Thank you very much for your kind review and feedback.

**Pull request merged**